nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/elasticsearch/ElasticsearchObject.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.rest.management.elasticsearch;
+
+import static javax.servlet.http.HttpServletResponse.SC_OK;
+import static org.nuxeo.ecm.core.api.security.SecurityConstants.SYSTEM_USERNAME;
+import static org.nuxeo.elasticsearch.bulk.IndexAction.ACTION_NAME;
+import static org.nuxeo.elasticsearch.bulk.IndexAction.INDEX_UPDATE_ALIAS_PARAM;
+
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+
+import org.apache.commons.lang3.StringUtils;
+import org.nuxeo.ecm.core.bulk.BulkService;
+import org.nuxeo.ecm.core.bulk.message.BulkCommand;
+import org.nuxeo.ecm.core.bulk.message.BulkStatus;
+import org.nuxeo.ecm.core.query.sql.NXQL;
+import org.nuxeo.ecm.webengine.model.WebObject;
+import org.nuxeo.ecm.webengine.model.impl.DefaultObject;
+import org.nuxeo.elasticsearch.api.ElasticSearchAdmin;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * Endpoint to administer Elasticsearch.
+ *
+ * @since 11.1
+ */
+@WebObject(type = "elasticsearch")
+public class ElasticsearchObject extends DefaultObject {
+
+    public static final String GET_ALL_DOCUMENTS_QUERY = "SELECT * from Document";
+
+    /**
+     * Performs an ES indexing on documents of the given repository and that match the given NXQL query.
```

Comment:
I would have put the `repositoryName` as `@param` in the Javadoc to have a shorter summary sentence.
Something like:
```suggestion
     * Performs an ES indexing on documents matching the optional NXQL query.
     *
     * @param repositoryName the repository name on which we launch the indexing, cannot be {@code null} or {@code empty}
```

Same thing for all other methods 😬